code/modules/projectiles/guns/ballistic/launchers.dm

```diff
@@ -57,55 +57,20 @@
 	casing_ejector = FALSE
 	weapon_weight = WEAPON_HEAVY
 	bolt_type = BOLT_TYPE_NO_BOLT
-	magazine_wording = "rocket"
+	internal_magazine = TRUE
+	cartridge_wording = "rocket"
 	empty_indicator = TRUE
 
-/obj/item/gun/ballistic/rocketlauncher/handle_atom_del(atom/A)
-	if(A == chambered)
-		chambered = null
-		if(!QDELETED(magazine))
-			QDEL_NULL(magazine)
-	if(A == magazine)
-		magazine = null
-		if(!QDELETED(chambered))
-			QDEL_NULL(chambered)
-	update_icon()
-	return ..()
-
-/obj/item/gun/ballistic/rocketlauncher/can_shoot()
-	return chambered?.BB
-
 /obj/item/gun/ballistic/rocketlauncher/unrestricted
 	pin = /obj/item/firing_pin
 
-/obj/item/gun/ballistic/rocketlauncher/process_chamber()
-	if(chambered)
-		chambered = null
-	if(magazine)
-		QDEL_NULL(magazine)
-	update_icon()
+/obj/item/gun/ballistic/rocketlauncher/afterattack()
+	. = ..()
+	magazine.get_round(FALSE) //Hack to clear the mag after it's fired
```

Comment:
does the icon still switch to the empty one after firing?

Reply:
Yes, the chamber is cleared (which is what the empty indicator is based on) and the contents of the magazine are actually a rocket that's been qdeleted. I didn't feel like going through the effort of writing a shitload of code to handle odd behavior on mag size 1 weapons firing caseless ammo with no bolt, so this works instead.